File: src/Python/209.MinimumSizeSubarraySum.py

class Solution:
    def minSubArrayLen(self, s, nums):
        n = len(nums)
        ans = 9999999999
        left = 0
        sum = 0
        for i in range(n):
            sum += nums[i]
            while sum >= s:
                ans = min(ans, i + 1 - left)
                sum -= nums[left]
                left += 1
        return ans if ans != 9999999999 else 0 



if __name__ == "__main__":
    solution = Solution()
    print(solution.minSubArrayLen(7, [2,3,1,2,4,3]))

File: src/CSharp/209.MinimumSizeSubarraySum.cs

using System;
using System.Collections.Generic;

namespace PeefyLeetCode.MinimumSizeSubarraySum
{
public class Solution {
    public int MinSubArrayLen(int s, int[] nums) {
        int n = nums.Length;
        int ans = int.MaxValue;
        int left = 0;
        int sum = 0;
        for (int i = 0; i < n; i++) {
            sum += nums[i];
            while (sum >= s) {
                ans = Math.Min(ans, i + 1 - left);
                sum -= nums[left++];
            }
        }
        return (ans != int.MaxValue) ? ans : 0;
    }
}
}